OpenRA.Mods.Common/Scripting/Global/LightingGlobal.cs

```diff
@@ -0,0 +1,64 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Scripting;
+
+namespace OpenRA.Mods.Common.Scripting
+{
+	[ScriptGlobal("Lighting")]
+	public class LightingGlobal : ScriptGlobal
+	{
+		readonly IEnumerable<FlashPaletteEffect> flashPaletteEffects;
+		readonly GlobalLightingPaletteEffect lighting;
+		readonly bool hasLighting;
+
+		public LightingGlobal(ScriptContext context)
+			: base(context)
+		{
+			flashPaletteEffects = context.World.WorldActor.TraitsImplementing<FlashPaletteEffect>();
+			lighting = context.World.WorldActor.TraitOrDefault<GlobalLightingPaletteEffect>();
+			hasLighting = lighting != null;
+		}
+
+		[Desc("Controls the `FlashPaletteEffect` trait.")]
+		public void Flash(string type = null, int ticks = -1)
+		{
+			foreach (var effect in flashPaletteEffects)
+				if (effect.Info.Type == type)
+					effect.Enable(ticks);
+		}
+
+		public double Red
+		{
+			get { return hasLighting ? lighting.Red : 0d; }
```

Comment:
My idea was to indicate that this wasn't usable (getting a bogus value back).
But perhaps 1d would have been a better default.